pkgs/ffigen/lib/src/config_provider/config.dart

```diff
@@ -12,368 +14,694 @@ import 'config_types.dart';
 
-  static bool _shouldIncludeDefault(Declaration declaration) => false;
+  static const excludeAll = Unions(shouldInclude: _excludeAll);
 
-  /// Applies renaming and returns the result.
-  final String Function(Declaration declaration) rename;
+  static const includeAll = Unions(shouldInclude: _includeAll);
 
-  static String _renameDefault(Declaration declaration) =>
-      declaration.originalName;
+  static Unions include(Set<String> names) => Unions(
+    shouldInclude: (Declaration decl) => names.contains(decl.originalName),
+  );
+}
 
-  /// Applies member renaming and returns the result. Used for struct/union
-  /// fields, enum elements, function params, and ObjC
-  /// interface/protocol/category methods/properties.
-  final String Function(Declaration declaration, String member) renameMember;
+/// Configuration for typedefs.
+final class Typedefs extends Declarations {
+  /// If typedef of supported types(int8_t) should be directly used.
+  final bool useSupportedTypedefs;
 
-  static String _renameMemberDefault(Declaration declaration, String member) =>
-      member;
+  /// If enabled, unused typedefs will also be generated.
+  final bool includeUnused;
 
-  /// Whether a member of a declaration should be included. Used for ObjC
-  /// interface/protocol/category methods/properties.
-  final bool Function(Declaration declaration, String member)
-  shouldIncludeMember;
+  /// Stores typedef name to ImportedType mappings specified by user.
+  final List<ImportedType> typeMappings;
+
+  const Typedefs({
+    super.rename,
+    super.renameMember,
+    super.shouldInclude,
+    super.shouldIncludeMember,
+    super.shouldIncludeSymbolAddress,
+    this.useSupportedTypedefs = true,
+    this.includeUnused = false,
+    this.typeMappings = const <ImportedType>[],
+  });
 
-  static bool _shouldIncludeMemberDefault(
-    Declaration declaration,
-    String member,
-  ) => true;
+  static const Typedefs excludeAll = Typedefs(shouldInclude: _excludeAll);
 
-  const DeclarationFilters({
-    this.shouldInclude = _shouldIncludeDefault,
-    this.shouldIncludeSymbolAddress = _shouldIncludeDefault,
-    this.rename = _renameDefault,
-    this.renameMember = _renameMemberDefault,
-    this.shouldIncludeMember = _shouldIncludeMemberDefault,
+  static const Typedefs includeAll = Typedefs(shouldInclude: _includeAll);
+
+  static Typedefs include(Set<String> names) => Typedefs(
+    shouldInclude: (Declaration decl) => names.contains(decl.originalName),
+  );
+}
+
+/// Configuration for Objective-C interfaces.
+final class ObjCInterfaces extends Declarations {
+  /// If enabled, Objective C interfaces that are not explicitly included by
+  /// the [Declarations], but are transitively included by other bindings,
+  /// will be code-genned as if they were included. If disabled, these
+  /// transitively included interfaces will be generated as stubs instead.
+  final bool includeTransitive;
+
+  /// The module that the ObjC interface belongs to.
```

Comment:
nit: here and elsewhere we should standardize on how to spell the programing language:

- line 516 says "Objective-C"
- line 518 says "Objective C"
- line 524 says "ObjC"

Reply:
We need an automated solution for this issue: https://github.com/dart-lang/native/issues/2589